docs/proposals/20210526-cluster-class-and-managed-topologies.md

```diff
@@ -1084,33 +565,17 @@ One of the key points of this proposal is that cluster topologies are continuous
-in order to reconcile objects in the cluster with the corresponding templates in the ClusterClass.
-
-The reconciliation process enforces that all the fields in the template for which a value is defined are reflected in the object.
-
-In practice:
-- If a field has a value in the template, but the users/another controller changes the value for this field in the object, the reconcile
-  topology process will restore the field value from the template; please note that due to how merge patches works internally:
-    - In case the field is a map, the reconcile process enforces the map entries from the template, but additional map entries
-      are preserved.
-    - In case the field is a slice, the reconcile process enforces the slice entries from the template, while additional values
-      are deleted.
-
-Please note that:
-- If a field does not have a value in the template, but the users/another controller sets this value in the object,
-  the reconcile topology process will preserve the field value.
-- If a field does not exist in the template, but it exists only in the object, and the users/another controller
-  sets this field's value, the reconcile topology process will preserve it.
-- If a field is defined as `omitempty` but the field type is not a pointer or a type that has a built-in nil value, 
-  e.g a `Field1 bool 'json:"field1,omitempty"'`, it won't be possible to enforce the zero value in the template for the field, 
-  e.g `field1: false`, because the field is going to be removed and technically there is no way to distinguishing
-  unset from the zero value for that type.
-  NOTE: this is a combination not compliant with the recommendations in the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required)).
-- If a field is defined as `omitempty` and the field type is a pointer or a type that has a built-in nil value,
-  e.g a `Field1 *bool 'json:"field1,omitempty"'`, it will be possible to enforce the zero value in the template 
-  for the field, e.g `field1: false` (but it won't be possible to enforce `field1: null`).
+More specifically, the topology controller uses [Server Side Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) to write/patch topology owned objects;
+using SSA allows other controllers to co-author the generated objects.
+
+However, this requires providers to pay attention on lists that are co-owned by multiple controller, for example lists that are expected to contain values from  ClusterClass/Variables, 
+and thus managed by the CAPI topology controller, and values from the infrastructure provider itself, like e.g. subnets in CAPA.
+
+In this cases for ServerSideApply to work properly it is required to ensure the proper annotation exists on the CRD
+types definitions, like +MapType or +MapTypeKey.
```

Comment:
```suggestion
type definitions, like +MapType or +MapTypeKey.
```

Reply:
Can we add a link to https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy here as well?
Just easier to understand for folks with the reference